### ISSUE TITLE
Unrestricted spaces/newlines after ( and before ) in function literals?

### ISSUE THREAD

**Issue by van-de-bugger** [leafo/moonscript#192]
I am trying to use LDoc on my MoonScript sources. Using `@param` command for documenting function parameters is too talkative:

    class Matrix
        ...
        --- Get/set a value of matrix element.
        --- @param row Row index.
        --- @param col Column index.
        --- @param value A new value of element, optional.
        elem: ( row, col, value ) => ...

It requires `@param` command, all the parameter names are duplicated. LDoc can infer parameter names from code:

    class Matrix
        ...
        --- Get/set a value of matrix element.
        elem: ( 
            row,  -- Row index.
            col,  -- Column index.
            value -- A new value of element, optional.
        ) => ...

Such code perfectly parsed by LDoc, but moonscript fails with error

> Failed to parse:
> [##] >>    elem: (

However, MoonScript CHANGELOG.md says:

> You can put unrestricted whitespace/newlines immediately after an opening parenthesis, and immediately before closing parenthesis.

That was implemented in v0.3.0 (2015-2-28). Examples shown in CHANGELOG illustrate expressions and function calls. My question is: does this claim should cover function literals also?

**Comment by baileyparker** [comment]
Just ran into this issue in a slightly different context (version 0.3.1). When calling a function, if you wrap the arguments in parentheses and spread them across multiple lines (regardless of arrangement), you get an error.
I think the answer to @van-de-bugger's question should be yes, because this issue does not occur when you don't wrap the arguments in parentheses. Of course, the purpose of parentheses is to resolve ambiguity so they can be necessary (and it makes sense that there may be ambiguity in a function call spanning multiple lines).

For example:

```moonscript
print(1, 2, 3) -- works as expected
print 1, 2, 3  -- also works as expected

-- But problems arise when we split arguments over lines
print 1,
  2, 3   -- this works as expected

print 1,
  2,
  3 -- this does as well

-- But NONE of the following work (parser error)
print(1,
  2, 3)

print(1,
  2,
  3)

print(1,
  2,
  3
)

print(
  1, 2,
  3
)

print(
  1,
  2,
  3
)

-- But oddly enough, this does work
print(
  1, 2, 3
)
```

From the fact that the last of these works, I assume the intention was to allow the same whitespace rules apply to arguments wrapped in parens. Namely, everything in this example script should work.

Or am I misinterpreting the reference?

**Comment by leafo** [comment]
Fixed with d8830b17f3dafcdd6b0357d584bd296730ce5ea6
